LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan.NetworkServer/ConcentratorDeduplication.cs

```diff
@@ -86,6 +96,23 @@ internal static string CreateCacheKey(UpstreamDataFrame updf)
             return BitConverter.ToString(key);
         }
 
+        private static string CreateCacheKeyCore(JoinRequestFrame joinReq)
+        {
+            var totalBufferLength = JoinEui.Size + DevEui.Size + DevNonce.Size;
+            Span<byte> buffer = stackalloc byte[totalBufferLength];
```

Comment:
So in this case it's guaranteed that the length will not exceed 128?

Reply:
Correct, because all of the fields we are using (JoinEui, DevEui and DevNonce) have a fixed size: 8 + 8 + 2 bytes in total. Unless we change these primitives drastically of course. 

In the `CreateCacheKeyCore(UpstreamDataFrame)` we include also the payload in the buffer which has a dynamic size, that's why if it's smaller than 128 bytes we use the stack, otherwise the heap.